rviz_default_plugins/src/rviz_default_plugins/displays/image/ros_image_texture.cpp

```diff
@@ -227,6 +227,7 @@ bool ROSImageTexture::update()
 
   width_ = image->width;
   height_ = image->height;
+  stride_ = image->step;
```

Comment:
Those three variables need to be set before calling setFormatAndNormalizeDataIfNecessary() because they are used inside of it. This needs retesting after you move them.